biblio-iacr.el

```diff
@@ -0,0 +1,91 @@
+;;; biblio-iacr.el --- description -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2020 Lorenzo Martinico
+
+;; Author: Lorenzo Martinico <lzmartinico@gmail.com>
+;; URL: https://github.com/cpitclaudel/biblio.el
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+;;
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+;;
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;
+;; This file implements a backend for the for the `biblio' package (which see for more
+;; documentation).
+;;
+;;; Code:
+
+(require 'biblio-core)
+
+(defun biblio-iacr--forward-bibtex (metadata forward-to)
+  "Forward BibTeX for IACR entry METADATA to FORWARD-TO."
+    (funcall forward-to
+         (format "@misc{cryptoeprint:%s,\n    author = {%s},\n    title = {%s},\n    howpublished = {%s},\n    year = {%s},\n    note = {\\url{%s}},\n}"
+                 (biblio-alist-get 'eprintid metadata)
+                 (car (biblio-alist-get 'authors metadata))
+                 (biblio-alist-get 'title metadata)
+                 (biblio-alist-get 'container metadata)
+                 (biblio-alist-get 'year metadata)
+                 (biblio-alist-get 'url metadata))))
+
+(defun biblio-iacr--extract-interesting-fields (header body)
+  "Prepare a IACR search result, composed of HEADER and BODY, for display.
+HEADER contains the technical report id, which includes the year of publication.
+BODY includes the Title and Authors."
```

Comment:
I wonder if there's a more robust way to write this than using nth and caddr; it might actually read nicer with pcase?

Reply:
Have a look at 6c6a175, using pcase is much more expressive, but I don't know if it adds anything in terms of robustness (if the IACR DOM changes, it will still fail in unexpected ways - but maybe it will be easier to fix now)

Reply:
I like pcase a lot better — it makes it clear what structure is expected, which should make it clearer if we need a fix later.